arcgis-ios-sdk-samples/Layers/Add ENC exchange set/AddENCExchangeSetViewController.swift

```diff
@@ -0,0 +1,108 @@
+            subdirectory: "hydrography/hydrography"
+        )!
+        .deletingLastPathComponent()
+        // Set environment settings for loading the dataset.
+        let environmentSettings = AGSENCEnvironmentSettings.shared()
+        environmentSettings.resourceDirectory = hydrographyDirectory
+        // The SENC data directory is for temporarily storing generated files.
+        environmentSettings.sencDataDirectory = FileManager.default.temporaryDirectory
+        
+        encExchangeSet.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                var ENCLayers = [AGSENCLayer]()
+                let loadGroup = DispatchGroup()
+                // A reference to the last error occurred, if there is any.
+                var loadError: Error?
+                // Create a list of ENC layers and load each layer.
+                encExchangeSet.datasets.forEach { dataset in
+                    let layer = AGSENCLayer(cell: AGSENCCell(dataset: dataset))
+                    ENCLayers.append(layer)
+                    loadGroup.enter()
+                    layer.load { error in
+                        defer { loadGroup.leave() }
+                        loadError = error
+                    }
+                }
+                // Add layers to the map.
+                map.operationalLayers.addObjects(from: ENCLayers)
+                
+                // Zoom to the combined extent of all ENC layers after they are loaded.
+                loadGroup.notify(queue: .main) { [weak self] in
+                    if let completeExtent = AGSGeometryEngine.combineExtents(ofGeometries: ENCLayers.compactMap { $0.fullExtent }) {
+                        self?.mapView.setViewpoint(AGSViewpoint(targetExtent: completeExtent), completion: nil)
+                    } else if let error = loadError {
+                        self?.presentAlert(error: error)
+                    }
+                }
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["AddENCExchangeSetViewController"]
+        addENCExchangeSet(mapView: mapView)
+    }
+    
+    deinit {
+        // Remove all files in tmp folder.
+        DispatchQueue.global(qos: .utility).async {
+            let tmpURL = FileManager.default.temporaryDirectory
+            if let tmpDirectory = try? FileManager.default.contentsOfDirectory(
+                at: tmpURL,
+                includingPropertiesForKeys: nil,
+                options: .skipsHiddenFiles
+            ) {
+                tmpDirectory.forEach { file in
+                    try? FileManager.default.removeItem(atPath: file.path)
+                }
+            }
+        }
+    }
```

Comment:
Ah, the temporary items are being removed on deinit. Good! However, this is way more complicated than it needs to be. As mentioned earlier, a subdirectory of the temporary directory should be used. Then, the subdirectory itself can just be removed using `FileManager.removeItem(at:)`. As far as I can tell, there's no reason to do that asynchronously by dispatching to a different queue.